jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -702,7 +702,16 @@ public void addEventListener(EventListener listener)
         }
 
         if (listener instanceof ServletContextListener)
-            _servletContextListeners.add((ServletContextListener)listener);
+        {
```

Comment:
I think we should throw IllegalStateException here:  you should not be able to add a ServletContextListener _after_ the context has initialized, as this is contrary to the Servlet Spec. I know this is a jetty api, however if you're using a SCL, you're using the spec. 

Furthermore, the ServletContextListener is supposed to be called before any servlets, filters etc have been initialized,  whereas this change allows a ServletContextListener to be called at any old time maybe even in a shutdown sequence - who knows what could happen. 

I also do not think it's a good idea to do the listener initialization in multiple places: having it in 1 place makes the (already very complicated) initialization sequence very clear.

Reply:
Will try with ISE and see if we break anything....